### ISSUE TITLE
ObjectMapper PropertyNamingStrategy SNAKE_CASE

### ISSUE THREAD

**Issue by Developer-Matt** [stoicflame/enunciate#810]
Hi,

Is there any way yet to set a custom object mapper. I have seen other issues about this where users are trying to work with complex objects and require custom object mapper but i would just like to set the PropertyNamingStrategy to snake case so that all fields are converted to snake case.  The only alternative at the moment is annotating every field with @JsonProperty and that would be a mission.

objectMapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);

Is this something that can already be achieved and I'm just not finding it, or would this have to be implemented?
If it needs to be implemented would this be a difficult task, am willing to try myself if you could point me in the right direction.

Thanks

**Comment by stoicflame** [comment]
Hi.

Unfortunately, support for snake case isn't currently supported, but it should be pretty easy to implement. I'll track this as an enhancement.

You'd need to implement a new method on the `com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext` (e.g. `getPropertyNamingStrategy`) that pulls the property naming strategy out of the config:

```xml
<enunciate>
  <modules>
    <jackson propertyNamingStrategy="SNAKE_CASE"/>
```

Then you'd modify the `getName` method of `com.webcohesion.enunciate.modules.jackson.model.Member` to honor the property naming strategy.

This would have to be done for both the `jackson` module and the `jackson1` module.

**Comment by Developer-Matt** [comment]
Hi, 
Thanks for the response and the comments, I will see if I can get this working.

**Comment by Developer-Matt** [comment]
Hi,

Thanks to your instructions I have managed to implement the above, and it was pretty easy.
Unfortunately I could not do a fully successful build, as the Full JBoss test could not run due to restrictions on my machine.  I could also not do a clean and install as I don't have the GNUStep and other libraries installed.
But I did messily just drop the built Jackson jars into my maven repository and they where used when I generated the documentation for my project, and that worked.

All seems to generate fine for me.  When I set propertyNamingStrategy="SNAKE_CASE" in my enunciate.xml, the field names are converted as such.
I am still generating more documentation but so far for what I need it works fine.

I don't know if you would like the code, its only a couple files.  I don't have permission to create a branch and therefore pull requests, but if you would like me to send it to you just shout.

Thanks for awesome project.
